google-cloud-examples/src/main/java/com/google/cloud/examples/bigquery/snippets/dataset/DeleteDataset.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.examples.bigquery.snippets.dataset;
+
+// [START bigquery_delete_dataset]
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.DatasetId;
+
+public class DeleteDataset {
+
+  public static boolean deleteDataset(String projectId, String dataset, Boolean deleteContents) {
+    // Get an instance of the BigQuery service.
+    BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
+
+    DatasetId datasetId = DatasetId.of(projectId, dataset);
+
+    // You can choose to safely delete which will fail if the dataset is not empty, or a recursive
+    // delete.
+    Boolean deleted = false;
+    if (deleteContents) {
+      deleted = bigquery.delete(datasetId, BigQuery.DatasetDeleteOption.deleteContents());
```

Comment:
It's not super clear to me which is safe and which is recursive - perhaps update each to have it's own comment to clarify.